db scripts/deleteContactEventsByDateFunction.sql

```diff
@@ -0,0 +1,17 @@
+CREATE OR REPLACE FUNCTION public.delete_contact_events_before_date(earliest_date character varying, curr_investigation_id integer)
+ RETURNS void
+ LANGUAGE plpgsql
+AS $function$
+	begin
+		delete from public.person p
```

Comment:
Why do you delete person? You need to delete only an event and the "Cascading Delete" will delete person also.

Reply:
the cascade only works with two levels: deleting the contact event triggers a delete cascade on contacted person, but not on person itself, by adding a delete cascade on the person foreign key in the contacted_person table I'm able to perform this delete

Reply:
1. I don't think you're right. On delete cascade means that if a record in the parent table is deleted, then the corresponding records in the child table will automatically be deleted.

2. Ofcourse you need to add a delete cascaded to person FK on contacted_person table for automatically delete. Here you're deleting person "Manually".

Reply:
There is a function: delete_contact_event_function which does that, only the where is by the event id. Anyway, uf that still doesnt workm maybe you should alter the fk to be on delete cascade and chage it once and for all, and then you could just delete the person: alter table public.scores
drop constraint scores_gid_fkey,
add constraint scores_gid_fkey
   foreign key (gid)
   references games(gid)
   on delete cascade;

Reply:
Basically you're not right. On delete cascade means that if a record in the parent table is deleted, then the corresponding records in the child table will automatically be deleted.

But as I see in our DB design, there is a bit problem with the FK and it's not in hierarchy order so you do need to delete "Manually" the person records.

Reply:
@MayaSasson I did that already, I uploaded a script that adds a delete cascade to contacted_person, that's why I added a mutation that deletes a single contact_event to be used whenever we want to delete one